src/main/scala/com/lucidchart/open/relate/ColReader.scala

```diff
@@ -0,0 +1,76 @@
+package com.lucidchart.open.relate
+
+import java.nio.ByteBuffer
+import java.sql.ResultSet
+import java.util.{Date, UUID}
+import shapeless._
+
+class NullColumnException(col: String) extends Exception(s"Unexpected null value in column: $col")
+
+trait ColReader[A] { self =>
+  def read(col: String, row: SqlResult): Option[A]
```

Comment:
kind of, but columns are 1-based, just in case you actually copy/paste that code